apps/api/src/app/feature-flags/services/feature-flags.service.ts

```diff
@@ -0,0 +1,51 @@
+import { Injectable, Logger } from '@nestjs/common';
+
+import { LaunchDarklyService } from './launch-darkly.service';
+
+import { IFeatureFlagsService } from '../types';
+
+const LOG_CONTEXT = 'FeatureFlagsService';
+
+export enum FeatureFlagsProvidersEnum {
+  LAUNCH_DARKLY = 'LaunchDarkly',
+}
+
+const featureFlagsProviders = {
+  [FeatureFlagsProvidersEnum.LAUNCH_DARKLY]: LaunchDarklyService,
+};
+
+@Injectable()
+export class FeatureFlagsService {
+  private service: IFeatureFlagsService;
+
+  constructor() {
+    // TODO: In the future we can replace the object key here for an environment variable
+    const service = featureFlagsProviders[FeatureFlagsProvidersEnum.LAUNCH_DARKLY];
```

Comment:
If we want to let the user have the ability to choose a provider, maybe it will be a good idea to introduce an environment variable with the provider name e.g
FEATURE_FLAG_PROVIDER=LaunchDarkly.
And update the current environment variable key LAUNCH_DARKLY_SDK_KEY to a generic name like FEATURE_FLAG_SDK_KEY, that way the user could choose from existing sdk's a provide that he wants to use.

Reply:
Exactly is what I'd like to do in the future. 👍🏻